src/main.rs

```diff
@@ -259,7 +260,7 @@ fn main() -> crossterm::Result<()> {
         match state {
             State::Test(ref mut test) => {
                 if let Event::Key(key) = event {
-                    test.handle_key(key);
+                    test.handle_key(&config, key);
```

Comment:
The key map should be passed into the `Test` once when it's first created and not every time `handle_key` is called, so that `Test` remains a pure state machine with key events as inputs.